lib/ex_gram/plug.ex

```diff
@@ -0,0 +1,37 @@
+defmodule ExGram.Plug do
+  @behaviour Plug
+
+  import Plug.Conn
+  alias Plug.Conn
+
+  @impl true
+  def init(opts), do: opts
+
+  @impl true
+  def call(%Conn{method: "POST"} = conn, _) do
+    if "telegram" in conn.path_info do
+      handle_update(conn)
+    else
+      conn
+    end
+  end
+
+  def call(conn, _), do: conn
+
+  defp handle_update(conn) do
+    token_hash = token_hash(conn.path_info)
+
+    {:ok, body, conn} = Plug.Conn.read_body(conn)
+    {:ok, update} = Jason.decode(body, keys: :atoms)
```

Comment:
```suggestion
    {:ok, update} = ExGram.Encoder.decode(body, keys: :atoms)
```

And probable we would want to handle the decode, and if it fails handle it correctly